adjust ribbon style

osodebailar opened the pull request:
Only optical adjustment:

- ribbon default background color is now the material button color, because the ribbons have button functionality
- add ripple effect on touch, introduced by @Andries-Smit for date/time 
- add an icon to the left.

Merge if you like it.

![ribbon_default_warning_dark](https://user-images.githubusercontent.com/25795894/164411071-edd469e2-e8de-4f4a-80f5-bd92c16be59d.PNG)
![ribbon_default_warning_light](https://user-images.githubusercontent.com/25795894/164411073-ac947a5d-bbb8-4a5d-90d6-64cd77da1143.PNG)
![ribbon_warning_dark](https://user-images.githubusercontent.com/25795894/164411075-34a15c50-e178-40a0-b37f-6a7f931207b2.PNG)
![ribbon_warning_light](https://user-images.githubusercontent.com/25795894/164411076-6d51df04-fa3d-45cc-a1e4-fd94e8fcd151.PNG)

Andries-Smit commented:
@osodebailar why not make them actual buttons?

MilosKozak commented on app/src/main/java/info/nightscout/androidaps/plugins/general/overview/OverviewFragment.kt:
```diff
@@ -832,6 +833,8 @@ class OverviewFragment : DaggerFragment(), View.OnClickListener, OnLongClickList
         binding.activeProfile.setBackgroundColor(profileBackgroundColor)
         binding.activeProfile.setTextColor(profileTextColor)
+        val drawableLeft: Array<Drawable?> = binding.activeProfile.compoundDrawables
+        if (drawableLeft[0] != null) profileTextColor.let { drawableLeft[0]!!.setTint(it) }
```
drawableLeft[0]?.let { it.setTint(profileTextColor) }

MilosKozak replied:
the same bellow

osodebailar replied:
Oh yes thx, mine was a little bit to complicated

osodebailar replied:
simlified statement and former code

MilosKozak commented:
we use different icon for profile everywhere

osodebailar commented:
> 

Oh do Not See the other ones . Where I can find them ? 
and in the end , what icon would we take ?

osodebailar commented:
> @osodebailar why not make them actual buttons?

Because until now they are not buttons . And to Convert them to buttons means to adjust the whole code for background and so on .
Same as in date/ time select .

MilosKozak commented:
> > 
> 
> Oh do Not See the other ones . Where I can find them ? and in the end , what icon would we take ?

the same code is repeating multiple times

osodebailar commented:
> multiple

changed it to the settings profile icon
![ribbon_profile_icon_dark](https://user-images.githubusercontent.com/25795894/164445373-1fec8792-b330-40c2-9a30-dba86087e5ba.PNG)



> > > 
> > 
> > 
> > Oh do Not See the other ones . Where I can find them ? and in the end , what icon would we take ?
> 
> the same code is repeating multiple times

simplified the code in overviewfragment

osodebailar commented:
> we use different icon for profile everywhere

tokk the same as we have in settings for local profile

osodebailar commented:
@MilosKozak so far ready from my side .

swissalpine commented:
Can we hide the buttons for low resolution devices? Thanks

osodebailar commented:
@swissalpine but if we hide them , you lost function

Philoul commented:
@osodebailar Concerning profile, you put icon of local profile plugin but everywhere "profile" icon is the 5 pointed star (profile switch dialog, profile switch button in actions, automation, profile viewer, compare profile...)

we have 2 identical icons available:
`@drawable/ic_actions_profileswitch`
`@drawable/ic_home_profile`

I think this icon is better than local profile plugin icon for profile in overview (short click show profile viewer, long click profile switch, both dialogs has the same 5 pointed star in header)

osodebailar commented:
Ok . But a 5 star icon is not very intuitive for a profile . The connection 5 star to Profil is hard to understand

Philoul commented:
It's another question (should we change profile icon or not to have a better understanding?)
I think consistancy of the 2 icons (if accepted) is the most important here.

In my mind (but I don't know if it was the reason of this choice in the past) the 5 pointed star illustrate the 5 main items you adjust in a profile 🤔:
- Insulin
- IC
- ISF
- Basal
- Target

osodebailar commented:
@Philoul  change the icon to five star 
@swissalpine remove the ribbons if style is set to low res

@MilosKozak : if you are satisfied you can merge

file:///home/developer/Pictures/ribbon_fivestar_light.PNG

Philoul commented:
Why you created a new star icon (with a different shape) and a new target icon (not consistant with existing target icon used everywhere in aaps)?

If it's just the size, I can generate smaller icons with the same shape, but it's probably possible (and better) to use existing xml files and resize them by code or xml parameter in the layout...

when I saw that there were 2 drawing files for the same profile icon, my first idea was to merge them in one file in a dedicated PR to have only one drawing for one icon used everywhere and ensure consistancy...

osodebailar commented:
@Philoul i need the other icon because it’s much more smaller .
If you want to create the right one , yes it’s up to you and i thank you If you can send me .

Philoul commented:
Ok, I'll create the 2 icons with your size and will send it to you (mail or PR on your repo 😉)

osodebailar commented:
@Philoul thx .
Mail is the best I think: 
andreas.eggen@gmail.von

swissalpine commented:
@osodebailar Sorry, I edited the post a minute after writing: I didn't mean the ribbon, just the icon.

osodebailar commented on app/src/main/java/info/nightscout/androidaps/skins/SkinLowRes.kt:
```diff
@@ -70,6 +70,8 @@ class SkinLowRes @Inject constructor(private val config: Config) : SkinInterface
             }
             graphCard.layoutParams = paramGraph
+
+            loopLayout.visibility = GONE
```
I never see your comment .
@swissalpine you bring me to work …. 10 seconds of  coding for nothing 😂😂

swissalpine replied:
I'm so sorry! ;-)

osodebailar commented:
@MilosKozak 
ready to mergenow. 

And many thx to @Philoul for the icon work.
![ribbon_dark](https://user-images.githubusercontent.com/25795894/164764966-c70a7e98-8fd1-4e4b-baa4-2e76d9c1b78e.PNG)
